src/zarr/sync.py

```diff
@@ -114,6 +114,23 @@ def _get_loop() -> asyncio.AbstractEventLoop:
     return loop[0]
 
 
+async def _collect_aiterator(data: AsyncIterator[T]) -> tuple[T, ...]:
+    """
+    Collect an entire async iterator into a tuple
+    """
+    result = []
+    async for x in data:
+        result.append(x)
```

Comment:
asyncio.gather? Like above, not much point in having coroutines if we serially wait for them.

Reply:
I don't think we can use `asyncio.gather` here, because `AsyncGenerator` is not iterable. Happy to be corrected, since I don't really know asyncio very well.

Reply:
and for clarification, `_collect_aiterator` largely exists for convenience in testing, because I need some way to collect async generators when debugging with `pdb`. This function is not intended for use in anything performance sensitive.

Reply:
We should probably re-examine the use of async-iterators, though. If we can't gather() on them (seems to be true?), then they are the wrong abstraction since gather() is probably always what we actually want.

Reply:
On second thoughts, maybe I'm wrong - does `async for` schedule all the coroutines at one?? Should be easy to test.

Reply:
I don't think it schedules them all at once. in `[x async for x in async_generator]`, `x` is not an awaitable; it's already awaited. since the basic model of the generator is that it's a resumable, stateful iterator, I don't think we can schedule all the tasks at once.

Reply:
the idea with the generators is to a) support seamless pagination and b) support pipelining (`del_prefix` will be able to take advantage of this at some point).